src/stan_math_backend/Expression_gen.ml

```diff
@@ -418,9 +418,20 @@ and pp_compiler_internal_fn ad ut f ppf es =
   | Some FnReadData -> read_data ut ppf es
   | Some FnReadParam -> (
     match es with
-    | {Expr.Fixed.pattern= Lit (Str, base_type); _} :: dims ->
-        pf ppf "@[<hov 2>in__.%s(@,%a)@]" base_type (list ~sep:comma pp_expr)
-          dims
+    | {Expr.Fixed.pattern= Lit (Str, constraint_string); meta= emeta}
+      :: {Expr.Fixed.pattern= Lit (Int, n_constraint_args_str); _} :: args ->
+        let n_constraint_args = int_of_string n_constraint_args_str in
+        let constraint_args, dims = List.split_n args n_constraint_args in
+        let lp_expr = Expr.Fixed.{pattern= Var "lp__"; meta= emeta} in
+        let arg_exprs = constraint_args @ [lp_expr] @ dims in
+        let maybe_constraint, maybe_jacobian =
+          if String.is_empty constraint_string then ("", "")
+          else ("_" ^ constraint_string, ", jacobian__")
+        in
```

Comment:
I think up here for not having `lp__` for plain reads `deserializer.read<>(dims...)` you want to check if `constrain_string` is blank (or if constrain_args is size 0? either or) and if not then don't include `lp__` as an arg